backend/tests/v5/e2e/routes/teamspaces/projects/models/common/revisions.test.js

```diff
@@ -0,0 +1,478 @@
+		describe.each(generateTestData(modelTypes.DRAWING))('Drawing', runTest);
+	});
+};
+
+const testCreateNewRevision = () => {
+	describe('Create New Revision', () => {
+		const basicData = generateBasicData();
+		const { users, teamspace, project, models, drawRevisions, conRevisions } = basicData;
+
+		beforeAll(async () => {
+			await setupData(basicData);
+		});
+
+		const generateTestData = (modelType) => {
+			let model;
+			let modelWithNoRev;
+			let modelNotFound;
+
+			if (modelType === modelTypes.CONTAINER) {
+				model = models.conWithRev;
+				modelWithNoRev = models.conWithNoRev;
+				modelNotFound = templates.containerNotFound;
+			} else {
+				model = models.drawWithRev;
+				modelWithNoRev = models.drawWithNoRev;
+				modelNotFound = templates.drawingNotFound;
+			}
+
+			const generateParams = () => deleteIfUndefined({
+				key: users.tsAdmin.apiKey,
+				ts: teamspace,
+				projectId: project.id,
+				modelId: model._id,
+				modelType,
+				file: dwgModel,
+				tag: modelType === modelTypes.CONTAINER ? ServiceHelper.generateRandomString(10) : undefined,
+				statusCode: modelType === modelTypes.DRAWING ? statusCodes[0].code : undefined,
+				revCode: modelType === modelTypes.DRAWING ? ServiceHelper.generateRandomString(10) : undefined,
+			});
+
+			return [
+				['the user does not have a valid session', { ...generateParams(), key: null }, false, templates.notLoggedIn],
+				['the teamspace does not exist', { ...generateParams(), ts: ServiceHelper.generateRandomString() }, false, templates.teamspaceNotFound],
+				['the user is not a member of the teamspace', { ...generateParams(), key: users.nobody.apiKey }, false, templates.teamspaceNotFound],
+				['the user does not have access to the model', { ...generateParams(), key: users.noProjectAccess.apiKey }, false, templates.notAuthorized],
+				['the user is viewer', { ...generateParams(), key: users.viewer.apiKey }, false, templates.notAuthorized],
+				['the user is commenter', { ...generateParams(), key: users.commenter.apiKey }, false, templates.notAuthorized],
+				['the project does not exist', { ...generateParams(), projectId: ServiceHelper.generateRandomString() }, false, templates.projectNotFound],
+				['the model does not exist', { ...generateParams(), modelId: ServiceHelper.generateRandomString() }, false, modelNotFound],
+				['the model is of wrong type', { ...generateParams(), modelId: models.federation._id }, false, modelNotFound],
+				['the file is missing', { ...generateParams(), file: undefined }, false, templates.invalidArguments],
+				['the file has incorrect format', { ...generateParams(), file: image }, false, templates.unsupportedFileFormat],
+				['the file is valid', generateParams(), true],
+				['the file is valid with uppercase extension', { ...generateParams(), file: dwgModelUppercaseExt, modelId: modelWithNoRev._id }, true],
+				...(modelType === modelTypes.DRAWING ? [
```

Comment:
Can we add a few more tests around quota?

1. the file doesn't exceed the maximum file size allowed
2. the teamspace has sufficient quota to upload this file

This applies to both containers and drawings

Reply:
Can we also spllit these case specific tests to new arrays with variables and do `...variableName` for better readability